<?xml version="1.0"?>
<launch>

    <!-- Publishers -->
    <arg name='topic_to_port_motor'             default='port_motor_speed' />
    <arg name='topic_to_strbrd_motor'           default='strbrd_motor_speed' />
    <arg name='topic_to_port_bow_thruster'      default='port_bow_thruster_speed' />
    <arg name='topic_to_strbrd_bow_thruster'    default='strbrd_bow_thruster_speed' />
    <arg name='topic_to_remote_control_status'  default='remote_control_status' />

    <!-- Subscribers -->
    <arg name='topic_from_joystick'         default='joy' />
    <arg name='topic_from_autonomy_status'  default='autonomy_status' />

    <!-- Launch teleoperations controller -->
    <node pkg="wamv_control" type="teleop_controller.py" name="teleop_controller_node">
        <remap from='port_motor_speed'          to='$(arg topic_to_port_motor)' />
        <remap from='strbrd_motor_speed'        to='$(arg topic_to_strbrd_motor)' />
        <remap from='port_bow_thruster_speed'   to='$(arg topic_to_port_bow_thruster)' />
        <remap from='strbrd_bow_thruster_speed' to='$(arg topic_to_strbrd_bow_thruster)' />
        <remap from='remote_control_status'     to='$(arg topic_to_remote_control_status)' />
        <remap from='autonomy_status'           to='$(arg topic_from_autonomy_status)' />
        <remap from='joy'                       to='$(arg topic_from_joystick)' />
   </node>

</launch>
